### ISSUE TITLE
Ignore not existing columns in the ListBuilder

### ISSUE THREAD

**Issue by Prokyonn** [sulu/sulu#7207]
| Q | A
| --- | ---
| Sulu Version | 2.5

# Actual Behavior

When you have a page with multiple tabs with listviews, e.g. the `Automation` tab from the AutomationBundle and the `Data` tab from the FormBundle, the query parameters for sorting/filtering the lists are copied from one tab to the other. 
Therefore it can happen, that you are on the `Automation` tab, sort after `schedule` and then move to the `Data` tab, the request sent to the server has then `sortBy=schedule` which ends in a 500, because the `Data` listview configuration does not have a column `schedule`.

# Expected Behavior

The request doesn't end in a 500 and returns the data with default sorting, ignoring the query parameters that define non-existing columns.

# Steps to Reproduce

see above

# Possible Solutions

Ignore query parameters that reference not existing columns

**Comment by MrSrsen** [comment]
It looks like bug in the `vendor/sulu/sulu/src/Sulu/Component/Rest/RestHelper.php` class in the `initializeListBuilder` method. This condition:

```php
        $sortBy = $this->listRestHelper->getSortColumn();
        if (null != $sortBy) {
            $listBuilder->sort($fieldDescriptors[$sortBy], $this->listRestHelper->getSortOrder());
        }
```

should look like this:

```php
        $sortBy = $this->listRestHelper->getSortColumn();
        if (null !== $sortBy && \array_key_exists($sortBy, $fieldDescriptors)) {
            $listBuilder->sort($fieldDescriptors[$sortBy], $this->listRestHelper->getSortOrder());
        }
```

I copied the original code and fixed it in my own class and decorated the original like this:

```
    App\Sulu\RestHelper:
        decorates: sulu_core.rest_helper
        arguments:
            - '@sulu_core.list_rest_helper'
```

and it fixed my problem. Good-enough hotfix for now.

**Comment by alexander-schranz** [comment]
@MrSrsen Sounds like a good solution. Do you want to create a pull request for this?

**Comment by mamazu** [comment]
Alternatively we could also allow the `sort` to allow `null` as sorting then we could remove the if statement entirely.

**Comment by MrSrsen** [comment]
> @MrSrsen Sounds like a good solution. Do you want to create a pull request for this?

@alexander-schranz to which repository should I make the PR? `sulu/sulu`? And shouldn't this rather be fixed in the JavaScript?

**Comment by mamazu** [comment]
Yes, `sulu/sulu` is the correct repository and I think that doing it in PHP is the better bet. Because we don't have the column information in Javascript.

**Comment by mamazu** [comment]
@Prokyonn By the way the same error also is present in the `ReferenceController` although it is not using the Doctrine Listener but rather its own implementation.
